Microservices.PlatformService/Controllers/PlatformController.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using AutoMapper;
+using Microservices.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microservices.PlatformService.Dtos;
+using Microservices.PlatformService.SyncDataServices.Http;
+using Microservices.PlatformService.SyncDataServices.RabbitMQ;
+
+namespace Microservices.PlatformService.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class PlatformController : ControllerBase
+    {
+        private readonly ICommandDataClient _commandDataClient;
+        private readonly IMessageBusClient _messageBusClient;
+        private readonly IUnitOfWorkService<PlatformReadDto, PlatformCreateDto> _service;
+        private readonly IMapper _mapper;
+
+        public PlatformController(IUnitOfWorkService<PlatformReadDto, PlatformCreateDto> service,
+            ICommandDataClient commandDataClient, IMessageBusClient messageBusClient, IMapper mapper)
+        {
+            _service = service;
+            _commandDataClient = commandDataClient;
+            _messageBusClient = messageBusClient;
+            _mapper = mapper;
+        }
+
+        [HttpGet("GetAll")]
+        public async Task<ActionResult<IEnumerable<PlatformReadDto>>> GetPlatforms()
+        {
+            var platforms = await _service.GetAllAsync();
+
+            return Ok(platforms);
+        }
+
+        [HttpGet("{id:int}", Name = "GetPlatformById")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<PlatformReadDto>> GetPlatformById(int id)
+        {
+            PlatformReadDto platform;
+            try
+            {
+                platform = await _service.GetByIdAsync(id);
+            }
+            catch (Exception e)
+            {
+                return e switch
+                {
+                    ArgumentNullException => NotFound(
```

Comment:
It would be better to use own Exception types for each service.

Reply:
Agree, but I decided not to use it in this project so as not to waste too much time.